File: test/API/example_hidraIngest.c
//
#include "hidraIngest.h"
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
//#include <unistd.h> //needed if sleep is used

int main()
{
    hidraIngest *obj;

    int chunksize=524288; //1024*512
    int rc;
    char source_file[25] = "/opt/hidra/test_file.cbf";
    FILE *fp;
    char *buffer = malloc(chunksize);
    int bytesRead;
    int i = 0;

    printf ("source_file %s\n", source_file);


    rc = hidraIngest_init (&obj);
    if (rc) exit(-9);

    fp = fopen(source_file,"rb"); // read mode
    assert(fp != NULL);

    rc = hidraIngest_createFile (obj, "test/test_file.cbf");
    if (rc)
    {
        perror ("createFile failed");
        rc = hidraIngest_stop(obj);
        assert(rc==0);
        exit(-9);
    }

    // read up to sizeof(buffer) bytes
    while ((bytesRead = fread(buffer, 1, chunksize, fp)) > 0)
    {
        printf ("The content of file %s:\n", source_file);
        printf ("Read file content of size: %i\n", bytesRead);
        rc = hidraIngest_write (obj, buffer, bytesRead);
        i++;
    }

    fclose(fp);

//    sleep(5);
    rc = hidraIngest_closeFile (obj);

    printf ("Stopping\n");
    rc = hidraIngest_stop(obj);

    free (buffer);

    return 0;


};

File: test/API/example_hidraIngest.py
from __future__ import print_function
# from __future__ import unicode_literals

import os
import zmq
import logging

from __init__ import BASE_DIR
import utils

from hidra import Ingest


# enable logging
logfile_path = os.path.join(BASE_DIR, "logs")
logfile = os.path.join(logfile_path, "test_ingest.log")
utils.init_logging(logfile, True, "DEBUG")

print("\n==== TEST: Ingest ====\n")

source_file = os.path.join(BASE_DIR, "test_file.cbf")
chunksize = 524288

context = zmq.Context()

obj = Ingest(use_log=True, context=context)

obj.create_file(os.path.join("test", "1.h5"))

# for i in range(5):
#    try:
#        data = "asdfasdasdfasd"
#        obj.write(data)
#        print "write"

#    except:
#        logging.error("break", exc_info=True)
#        break

# Open file
source_fp = open(source_file, "rb")
print("Opened file:", source_file)

while True:
    try:
        # Read file content
        content = source_fp.read(chunksize)
        logging.debug("Read file content")

        if not content:
            logging.debug("break")
            break

        obj.write(content)
        logging.debug("write")

    except:
        logging.error("break", exc_info=True)
        break

# Close file
source_fp.close()
logging.debug("Closed file: {0}".format(source_file))


try:
    obj.close_file()
except:
    logging.error("Failed to close file", exc_info=True)

logging.info("Stopping")

obj.stop()

print("\n==== TEST END: Ingest ====\n")
